examples/Building-and-deploying-multi-stage-RecSys/02-Deploying-multi-stage-RecSys-with-Merlin-Systems_filtering.ipynb

```diff
@@ -0,0 +1,738 @@
+      "  ValueType.FLOAT: (np.float, False, False),\n"
+     ]
+    }
+   ],
+   "source": [
+    "from merlin.systems.dag.ops.feast import QueryFeast \n",
+    "\n",
+    "user_features = [\"user_id_raw\"] >> QueryFeast.from_feature_view(\n",
+    "    store=feature_store,\n",
+    "    view=\"user_features\",\n",
+    "    column=\"user_id_raw\",\n",
+    "    include_id=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27e25be7-3ff0-49c2-a3fc-03ec4d615e77",
+   "metadata": {},
+   "source": [
+    "Retrieve top-K candidate items using `retrieval model` that are relevant for a given user. We use `PredictTensorflow()` operator that takes a tensorflow model and packages it correctly for TIS to run with the tensorflow backend."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "21139caa-3a51-42e6-b006-21a92c95f1bc",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<function tensorflow.python.dlpack.dlpack.from_dlpack(dlcapsule)>"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# prevent TF to claim all GPU memory\n",
+    "from merlin.models.loader.tf_utils import configure_tensorflow\n",
+    "\n",
+    "configure_tensorflow()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "47c2d9b1-51dc-4549-977d-d7941ee6486c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-08-01 20:01:10.368513: I tensorflow/core/platform/cpu_feature_guard.cc:152] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX\n",
+      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
+      "2022-08-01 20:01:11.486337: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 16249 MB memory:  -> device: 0, name: Quadro GV100, pci bus id: 0000:2d:00.0, compute capability: 7.0\n",
+      "08/01/2022 08:01:13 PM WARNING:No training configuration found in save file, so the model was *not* compiled. Compile it manually.\n"
+     ]
+    }
+   ],
+   "source": [
+    "topk_retrieval = 100\n",
+    "retrieval = (\n",
+    "    user_features\n",
+    "    >> PredictTensorflow(retrieval_model_path)\n",
+    "    >> QueryFaiss(faiss_index_path, topk=topk_retrieval)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "ad1de211-23a2-4fa6-8c1c-61b626e0cb3b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filter out anything that was in the user's current session\n",
+    "filtering = retrieval[\"candidate_ids\"] >> FilterCandidates(\n",
+    "    filter_out=user_features[\"item_id_seen\"]\n",
```

Comment:
Try making this `filter_out=user_features["item_id_seen_1"]`?

Reply:
still doesnt work unfortunately.